CONTRIBUTING.rst

```diff
@@ -0,0 +1,333 @@
+
+--------------
+
+Creating a Pull Request
+-----------------------
+
+Creating a pull request is fairly simple, just make sure it focuses on a
+single aspect and doesn’t manage to have scope creep and it’s probably
+good to go.
+
+Formatting
+~~~~~~~~~~
+
+We would greatly appreciate the code submitted to be of a consistent
+style with other code in disnake. This project follows PEP-8 guidelines
+(mostly) with a column limit of 100 characters.
+
+We use `PDM <https://pdm.fming.dev/>`__ for development. If PDM is
+not already installed on your system, you can follow their `installation
+steps here <https://pdm.fming.dev/latest/#installation>`__ to get
+started.
+
+Once PDM is installed and avaliable, use the following command to
+initialise a virtual environment, install the necessary development
+dependencies, and install the
+`pre-commit <https://pre-commit.com/#quick-start>`__ hooks. .
+
+::
+
+   pdm run setup_env
+
+The installed ``pre-commit`` hooks will automatically run before every
+commit, which will format/lint the code to match the project’s style.
+Note that you will have to stage and commit again if anything was
+updated!
+
+Most of the time, running pre-commit will automatically fix any issues
+that arise, but this is not always the case. We have a few hooks that
+*don’t* resolve their issues automatically, and must be fixed manually.
+One of these is the license header, which must exist in all files unless
+comments are not supported in those files, or they are not text files,
+in which case exceptions can be made. These headers must exist following
+the format documented at https://spdx.dev/ids/.
+
+Scripts
+~~~~~~~
+
+To run all important checks and tests, use ``pdm run nox``:
+
+.. code:: sh
+
+   pdm run nox -R
+
+You can also choose to only run a single task; run ``pdm run --list`` to
+view all available scripts and use ``pdm run <name>`` to run them.
+
+| Some notes (all of the mentioned scripts are automatically run by
+  ``pdm run nox -R``, see above): - If ``pre-commit`` hooks aren’t
+  installed, run ``pdm run lint`` manually to check and fix the
+  formatting in all files.
+| **Note**: If the code is formatted incorrectly, ``pre-commit`` will
+  apply fixes and exit without committing the changes - just stage and
+  commit again. - For type-checking, run ``pdm run pyright``. You can
+  use ``pdm run pyright -w`` to automatically re-check on every file
+  change.
+| **Note**: If you’re using VSCode and pylance, it will use the same
+  type-checking settings, which generally means that you don’t
+  necessarily have to run ``pyright`` separately. However, there can be
+  version differences which may lead to different results when later run
+  in CI on GitHub. - Tests can be run using ``pdm run test``. If you
+  changed some functionality, you may have to adjust a few tests - if
+  you added new features, it would be great if you added new tests for
+  them as well.
```

Comment:
Unsure what happened here - in any case, it doesn't render correctly:
![image](https://user-images.githubusercontent.com/8530778/232253060-ffc57d07-0552-458a-9ad1-629dddea15e9.png)